### ISSUE TITLE
Page not loading

### ISSUE THREAD

**Issue by yukixo** [kodjodevf/mangayomi#228]
### Steps to reproduce

Open series, open episode. Certain pages do not load even if you reload. This happens on the same pages every time. 

### Expected behavior

The pages should load

### Actual behavior

The page does not load, and reloading does nothing

### Mangayomi version

0.2.75 and previous versiom

### Device

Windows 10

### Other details

I have updated extension

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/kodjodevf/mangayomi-extensions/issues/new/choose).
- [X] I have updated all installed extensions.

**Comment by mumiby** [comment]
Hello maybe try another source (extension). The source's script can be broken because of some changes on the source website/server. If another source works then this issue should be reported in the extensions repository (link in your post).

**Comment by yukixo** [comment]
Ive tried using different manga apps using the same extension and it worked fine. I only have the reload page bug on mangayomi. It happens on android and windows 10 in the same pages, only on mangayomi, using the same extension that i do on other apps

**Comment by mumiby** [comment]
First, what's the name of the extension you have issue with? I'll try to replicate.
Second, does this issue happens with another extension (on Mangayomi)?
Then, is this issue specific to one manga/manhwa or does it affect all the content display by that extension?

**Comment by mumiby** [comment]
I tried and I noticed that some of page you pointed to don't load, just as you said:
- Rent-a... Chap 6 official pages 11 and 12 don't load although on the website (webview option)  they do on both android and windows (i think I saw a 403 error there).
- Same thing for Chap 31 asurascans page 6 and 7 
- Same thing for Rent-a.... Chap 1 page 16 on Mangadex. 
But there are other versions of theses chapters. Just use read them instead or use the webview option or switch to another source (they're plenty) The 403 error is "unable to bypass Cloudfare" error

**Comment by yukixo** [comment]
Yes and i really appreciate mangayomi. It is my favorite way to enjoy manga on both phone and computer. This is the only issue i have had so far and i thought it was significant enough to make a bug report. Just like you said, when i open the webview, the pages load normally. I just wanted to bring this issue to attention so that it could be looked into. Thank you for your time it is very appreciated.

**Comment by Nepalibabuu** [comment]
Mine, it doesn't let me read downloaded manga and shows connection exception 
![Screenshot_2024-10-01-20-46-04-670_com kodjodevf mangayomi](https://github.com/user-attachments/assets/0d9c4e47-771a-4647-a841-37d25d1daa28)

**Comment by ankhzet** [comment]
@kodjodevf, @yukixo Found the reason.

The image url's on Comick can have underscores (`_`) in the filename in (almost) any position, i.e. `23-abcef-ab_c.jpg`. Sometimes (quite often, it seems) it happens to be in the last position in the filename, i.e. `19--QEYm7tAOoND_.jpg`.
The problem is that Comick extension uses the `_.` as a separator when it parses the image url's in the chapter json. It first collects all url's in a single string joined by the separator, and then splits result by it to get the final array:
```dart
sonPathToString(res, r'$.chapter.images[*].url', '_.').split('_.')
```

Examle of the offending url:
https://meo.comick.pictures/19--QEYm7tAOoND_.jpg
It get's split into two urls, resulting in exactly two image load errors: 
```
https://meo.comick.pictures/19--QEYm7tAOoND
jpg
```
Similar problem is likely to occur in other extensions.
The solution can be to use unique separators, or `\0` null byte - it is unlikely to occur, naturally in html/json, ever.

I've temporarily fixed it in [my code](https://github.com/ankhzet/mangayomi/commit/e49ce00c70b7275cc9ef9f319ff442529e06992d#diff-6a6d8635d5ee6fd60730a098666bc2becad62b29f01265ac64a1f207b965c368R36) by patching the results of the parse, just fixing the extension source does not fix the issue with cached parse results (`ChapterPageurls` embedded in settings document)
